src/orange/surf/detail/InvoluteSolver.hh

```diff
@@ -0,0 +1,313 @@
+{
+    CELER_EXPECT(r_b > 0);
+    CELER_EXPECT(a >= 0);
+    CELER_EXPECT(tmax > 0);
+    CELER_EXPECT(tmin >= 0);
+    CELER_EXPECT(tmax < 2 * constants::pi + tmin);
+}
+
+//---------------------------------------------------------------------------//
+
+/*!
+ * Find all roots for involute surfaces that are within the bounds and result
+ * in positive distances. Performed by doing a Regular Falsi Iteration on the
+ * root function, \f[
+ * f(t) = r_b * [v{cos(a+t) + tsin(a+t)} + u{sin(a+t) - tcos(a+t)}] + xv - yu
+ * \f]
+ * where the Regular Falsi Iteration is given by: \f[
+ * t_c = [t_a*f(t_b) - t_b*f(t_a)] / [f(t_b) - f(t_a)]
+ * \f]
+ * where \em t_c replaces the bound with the same sign (e.g. \em t_a and \em
+ * t_b
+ * ). The initial bounds can be determined by the set of: \f[
+ * {0, beta - a, beta - a - pi, beta - a + pi, beta - a - 2pi, beta - a + 2pi
+ * ...} /f] Where \em beta is: \f[ beta = arctan(-v/u) \f]
+ */
+CELER_FUNCTION auto
+InvoluteSolver::operator()(Real3 const& pos,
+                           Real3 const& dir,
+                           SurfaceState on_surface) const -> Intersections
+{
+    using constants::pi;
+    // Flatten pos and dir in xyz and uv respectively
+    real_type x = pos[0];
+    real_type const y = pos[1];
+
+    real_type u = dir[0];
+    real_type v = dir[1];
+
+    // Mirror systemm for counterclockwise involutes
+    if (sign_)
+    {
+        x = -x;
+        u = -u;
+    }
+
+    /*
+     * Define tolerances.
+     * tol_point gives the tolerance level for a point,
+     * account for the floating point error when performing square roots.
+     * tol_conv gives the tolerance for the Regular Falsi iteration.
+     */
+    real_type tol_conv = r_b_ * tol();
+    real_type tol_point = r_b_ * tol() * 100;
+
+    // Results initalization and root counter
+    Intersections result;
+    int j = 0;
+    // Initial result vector.
+    result = {no_intersection(), no_intersection(), no_intersection()};
+
+    // Return result if particle is travelling along z-axis.
+    if (u == 0 && v == 0)
+    {
+        return result;
+    }
+
+    // Conversion constant for 2-D distance to 3-D distance
+
+    real_type convert = 1 / std::sqrt(ipow<2>(v) + ipow<2>(u));
+    u *= convert;
+    v *= convert;
+
+    // Remove 0 dist if particle is on surface
+    if (on_surface == SurfaceState::on)
+    {
+        result[j] = 0;
+        j++;
+    }
+
+    // Line angle parameter
+
+    real_type beta = line_angle_param(u, v);
+
+    // Setting first interval bounds, needs to be done to ensure roots are
+    // found
+    real_type t_lower = 0;
+    real_type t_upper = beta - a_;
+
+    // Round t_upper to the first positive multiple of pi
+    real_type floor_upper = -std::floor(t_upper / pi);
+    t_upper += max<real_type>(0, floor_upper) * pi;
+
+    // Parameters that will be used in loop
+    int i = 1;
+
+    // Lambda used for calculating the roots using Regular Falsi Iteration
```

Comment:
Regular -> Regula here and below
```suggestion
    // Parameters that will be used in loop
    int i = 1;

    // Lambda used for calculating the roots using Regula Falsi Iteration
```
Instead of "parameters" (not very descriptive) can you add exactly what `i` is?